src/core/moves/MetropolisHastingsMove.cpp

```diff
@@ -320,11 +424,34 @@ void MetropolisHastingsMove::performMcmcMove( double prHeat, double lHeat, doubl
         {
             if ( the_node->isClamped() )
             {
-                ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                try
+                {
+                    ln_likelihood_ratio += the_node->getLnProbabilityRatio();
+                }
+                catch (RbException e)
+                {
+                    ln_likelihood_ratio = RbConstants::Double::neginf;
+
+                    if ( e.getExceptionType() != RbException::MATH_ERROR )
+                    {
+                        throw RbException(e);
```

Comment:
throw;

Reply:
Same as above; does "throw;" actually propagate the error message?

Reply:
Surprisingly it does! 
It re-throws the exception object that was caught, without making a copy.

Reply:
Changed with previous commit.